File: apps/web/lib/firebase/forum-audience.ts
import { doc, getDoc } from "firebase/firestore";
import { getFirebaseDb } from "./client";

/**
 * Active platform audience for proportional Spotlight (excludes deactivated /
 * pending-deletion accounts). Backed by `platformStats/overview`.
 */
export async function fetchForumAudienceSize(): Promise<number> {
  const snap = await getDoc(doc(getFirebaseDb(), "platformStats", "overview"));
  if (!snap.exists()) return 0;
  const data = snap.data() as Record<string, unknown>;
  const total = Number(data.totalUsers ?? 0) || 0;
  const deactivated = Number(data.deactivated ?? 0) || 0;
  const pendingDeletion = Number(data.pendingDeletion ?? 0) || 0;
  return Math.max(0, total - deactivated - pendingDeletion);
}

File: apps/mobile/lib/features/forums/forum_audience.dart
import 'package:cloud_firestore/cloud_firestore.dart';

/// Active platform audience for proportional Spotlight (excludes deactivated /
/// pending-deletion accounts). Backed by `platformStats/overview`.
Future<int> fetchForumAudienceSize({FirebaseFirestore? firestore}) async {
  final db = firestore ?? FirebaseFirestore.instance;
  final snap = await db.collection('platformStats').doc('overview').get();
  if (!snap.exists) return 0;
  final data = snap.data() ?? const <String, dynamic>{};
  final total = (data['totalUsers'] as num?)?.toInt() ?? 0;
  final deactivated = (data['deactivated'] as num?)?.toInt() ?? 0;
  final pendingDeletion = (data['pendingDeletion'] as num?)?.toInt() ?? 0;
  final size = total - deactivated - pendingDeletion;
  return size < 0 ? 0 : size;
}
